libbindgen/src/ir/enum_ty.rs

```diff
@@ -49,8 +49,11 @@ impl Enum {
         }
 
         let declaration = ty.declaration().canonical();
-        let repr = Item::from_ty(&declaration.enum_type(), None, None, ctx)
-            .ok();
+        let et = declaration.enum_type();
+        if et.is_none() {
+            return Err(ParseError::Continue);
+        }
+        let repr = Item::from_ty(&et.unwrap(), None, None, ctx).ok();
```

Comment:
Looks like the test still fails (probably because of the early return). Have you tried what @emilio had suggested earlier?

``` rust
let repr = declaration.enum_type().and_then(|ty| Item::from_ty(&ty, None, None, ctx).ok());
```

This will chain both the options.